src/Query/index.ts

```diff
@@ -10,6 +10,49 @@ function processPart(part: string): string[] {
   return tokenize(part, false);
 }
 
+const BURIED_PROPERTIES = ['realm'];
+const BURIED_FACTOR = 0.8;
+
+//check type of parts
+function constructAgg(parts: any[]): object[] {
+  const newParts: any[] = [];
+  for (var part of parts) {
```

Comment:
(Nit) Can we use `let` or `const` instead of `var`? Just want to avoid unnecessary hoisting